File: Java/src/main/java/QuestionBank/Solution1001_1100/Solution1013.java
package QuestionBank.Solution1001_1100;
/**
 * @author windows
 */
public class Solution1013 {
    public boolean canThreePartsEqualSum(int[] arr) {
        int sum = 0;
        for (int i = 0; i < arr.length; i++) {
            sum += arr[i];
        }
        if (sum % 3 != 0){
            return false;
        }
        int count = 0;
        int sumTemp = 0;
        for (int i = 0; i < arr.length; i++) {
            sumTemp += arr[i];
            if (sumTemp == sum / 3){
                count++;
                sumTemp = 0;
            }
        }
        if (sum == 0){
            return count >= 2;
        }
        return count == 3;
    }
    
    
    public static void main(String[] args) {
        Solution1013 solution1013 = new Solution1013();
    }
}

File: C++/src/Solution1001_1100/Solution1013.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1013.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool canThreePartsEqualSum(vector<int>& arr) {
  int n = arr.size();
  int sum = 0;
  for (const auto &item : arr){
    sum += item;
  }
  if (sum % 3 != 0){
    return false;
  }
  int left = 0;
  int num = 0;
  int count = 0;
  for (const auto &item : arr){
    num += item;
    left++;
    if (num == sum / 3){
      count++;
      num = 0;
      if (count == 2){
        break;
      }
    }
  }
  return count == 2;
}